Design/new-mangling-approach.md

```diff
@@ -0,0 +1,74 @@
+## Name Mangling
+
+### 1. Registration of Types
+
+- Types are registered to track unique identifiers for variables.
```

Comment:
If you just say `type`, the assumption is that it's about Kotlin types.  Explain the structure of names.